db/osqlcomm.c

```diff
@@ -6743,6 +6743,10 @@ int osql_process_packet(struct ireq *iq, unsigned long long rqid, uuid_t uuid,
                 iq->usedb = iq->sc->db;
             rc = finalize_schema_change(iq, iq->sc_tran);
             iq->usedb = NULL;
+            if (iq->sc_locked) {
+                unlock_schema_lk();
+                iq->sc_locked = 1;
```

Comment:
This is wrong. The schema change is being committed and the schemalk will be released after it commits/aborts. See the comment above. There might be bugs there, but I dont think releasing lock here is correct.

Reply:
Yep, this was definitely wrong. Thanks